### DISCUSSION TITLE
Tag multiple voices

### DISCUSSION THREAD

**Discussion by jinsu35** [openai/whisper#265] [Ideas]
I tried using whisper to transcribe a conversation between three people and I thought it would be great if we could tag different voices. So the final result is a list of texts each with the speaker's id in front of it. For example "Alice: Roses are red..." "Bob: But not all roses are red" and so on.

**Comment by jongwook** [comment]
It is not supported out-of-the-box, but it has been tried in #264 to combine pyannote's diarization with Whisper.

**Reply by elmollero** [reply]
Buenos días

**Comment by Majdoddin** [comment]
@jinsu35 @jongwook 
[www.lexicaps.com](http://www.lexicaps.com/) seamlessly adds diarization to Whispers transcription. No 3rd party packages.
Feel free to contact me if you need more features.
Announcement: https://github.com/openai/whisper/discussions/1537
Repo: https://github.com/Majdoddin/lexicaps

**Comment by brainwane** [comment]
I noticed [someone used whisper.cpp with `tinydiarize` to do this](https://mefiwiki.com/wiki/Podcast_193_Transcript).

> whisper.cpp 1.4.0, large model, small tinydiarize model

**Reply by mrienstra** [reply]
That model is available here: https://huggingface.co/akashmjn/tinydiarize-whisper.cpp/tree/main

It can be downloaded within [`whisper.cpp`](https://github.com/ggerganov/whisper.cpp) with this command:
```bash
bash ./models/download-ggml-model.sh small.en-tdrz
```
... thanks to @akashmjn's PR (merged Jul 3, 2023): https://github.com/ggerganov/whisper.cpp/pull/1058

**Comment by toanhuynhnguyen** [comment]
Try WhisperX!
